### ISSUE TITLE
hnix executable should use consistent layout options

### ISSUE THREAD

**Issue by sjakobi** [haskell-nix/hnix#688]
In different parts of the code, different `prettyprinter` `LayoutOptions` are used:

Most parts use the [`defaultLayoutOptions`](http://hackage.haskell.org/package/prettyprinter-1.7.0/docs/Prettyprinter-Internal.html#v:defaultLayoutOptions) with [ribbon width](http://hackage.haskell.org/package/prettyprinter-1.7.0/docs/Prettyprinter.html#t:PageWidth) 1.0. Some parts of the code use this setting implicitly, for example via `Doc`'s `Show` instance:

https://github.com/haskell-nix/hnix/blob/778edf2c38b92f638b793a3a66a7735e46747b0f/main/Main.hs#L158-L163

Ribbon width 0.4 is also in use though:

https://github.com/haskell-nix/hnix/blob/778edf2c38b92f638b793a3a66a7735e46747b0f/main/Main.hs#L135-L141

https://github.com/haskell-nix/hnix/blob/778edf2c38b92f638b793a3a66a7735e46747b0f/tests/ParserTests.hs#L394-L402

Unless someone's interested in investigating whether ribbon width 0.4 results in more "beautiful" output, I'd suggest standardizing on ribbon width 1.0, i.e. `defaultLayoutOptions`.

**Comment by Anton-Latukha** [comment]
I vote for rustic unification.

Beautification, always can happen down the line by the appearance of a person who decides to do make it.

**Comment by rycee** [comment]
I think it would be great if hnix formatted code in the same style as [nixfmt](https://github.com/serokell/nixfmt/). Perhaps it would even be possible to use nixfmt as a library to do the formatting?

**Comment by Anton-Latukha** [comment]
Yes, implementation of `nixfmt` seems good, we should try to integrate with it somehow.

**Comment by Anton-Latukha** [comment]
I'd opted for `layoutSmart`.

The default of `defaultLayoutOptions` seems strange to me.

For a number of years, the terminals are not restricted to 80 & have dynamically changing width most of the time.

So for me it is strange the library does not determine, not have provided the ability, to make the output fit the actual terminal.

Let us use the `terminal-size` for it in all cases & `1` ribbon & we can always fall back to 80chars if the terminal happens to be on some exotic OS (but in that case the proper path again - that autodetection needs to be present in the upstream library).

**Comment by Anton-Latukha** [comment]
Special thanks to mentioning the form of output in CLI output modes in the head post, my mind got boxed into the code forms.
